index.d.ts

```diff
@@ -1,255 +1,234 @@
+  }
+
+  export interface HttpArchiveRequest {
+    url?: string;
+    method?: string;
+    headers?: NameValuePair[];
+    postData?: {
+      mimeType?: string;
+      params?: NameValuePair[];
+    };
+  }
+
+  export interface NameValuePair {
+    name: string;
+    value: string;
+  }
+
+  export interface Multipart {
+    chunked?: boolean;
+    data?: {
+      'content-type'?: string,
+      body: string
+    }[];
+  }
+
+  export interface RequestPart {
+    headers?: Headers;
+    body: any;
+  }
+
+  export interface Request extends Stream {
+    readable: boolean;
+    writable: boolean;
+
+    getAgent(): Agent;
+    // start(): void;
+    // abort(): void;
+    pipeDest(dest: any): void;
+    setHeader(name: string, value: string, clobber?: boolean): Request;
+    setHeaders(headers: Headers): Request;
+    qs(q: Object, clobber?: boolean): Request;
+    form(): FormData;
+    form(form: any): Request;
+    multipart(multipart: RequestPart[]): Request;
+    json(val: any): Request;
+    aws(opts: AWSOptions, now?: boolean): Request;
+    auth(username: string, password: string, sendInmediately?: boolean, bearer?: string): Request;
+    oauth(oauth: OAuthOptions): Request;
+    jar(jar: CookieJar): Request;
+
+    on(event: string, listener: Function): this;
+    on(event: 'request', listener: (req: ClientRequest) => void): this;
+    on(event: 'response', listener: (resp: IncomingMessage) => void): this;
+    on(event: 'data', listener: (data: Buffer | string) => void): this;
+    on(event: 'error', listener: (e: Error) => void): this;
+    on(event: 'complete', listener: (resp: IncomingMessage, body?: string | Buffer) => void): this;
+
+    write(buffer: Buffer, cb?: Function): boolean;
+    write(str: string, cb?: Function): boolean;
+    write(str: string, encoding: string, cb?: Function): boolean;
+    write(str: string, encoding?: string, fd?: string): boolean;
+    end(): void;
+    end(chunk: Buffer, cb?: Function): void;
+    end(chunk: string, cb?: Function): void;
+    end(chunk: string, encoding: string, cb?: Function): void;
+    pause(): void;
+    resume(): void;
+    abort(): void;
+    destroy(): void;
+    toJSON(): Object;
+  }
+
+  export interface Headers {
+    [key: string]: any;
+  }
+
+  export interface AuthOptions {
+    user?: string;
+    username?: string;
+    pass?: string;
+    password?: string;
+    sendImmediately?: boolean;
+    bearer?: string;
+  }
+
+  export interface OAuthOptions {
+    callback?: string;
+    consumer_key?: string;
+    consumer_secret?: string;
+    token?: string;
+    token_secret?: string;
+    verifier?: string;
+  }
+
+  export interface HawkOptions {
+    credentials: any;
+  }
+
+  export interface AWSOptions {
+    secret: string;
+    bucket?: string;
+  }
+
+  export type CookieJar = toughCookie.CookieJar;
+  export type Cookie = toughCookie.Cookie;
 }
 
-declare var request: request.RequestAPI<request.Request, request.CoreOptions, request.RequiredUriUrl>;
```

Comment:
This declaration is still here but at the very top (from what I experienced when running the tests, in order to declare both a namespace and a variable with the same name, you have to declare the variable first. But you can try it yourself (place it bellow the namespace and then run `npm run build+test`).